File: Review-10-Python3-Shapes/Example-2/shapes/Square.py
import copy

from shapes.Shape import Shape


class Square(Shape):
    """
    A Rectangle with 4 Equal Sides
    """

    def __init__(self, side=1):
        """
        Construct a Square
        """
        self.name = "Square"

        self._side = side

    @property
    def side(self):
        return self._side

    @side.setter
    def side(self, s):
        self._side = s

    def area(self):
        """
        Compute the area

        :return: area
        """
        return self._side ** 2.0

    def perimeter(self):
        """
        Compute the perimeter

        :return: perimeter
        """

        return 4 * self._side

    def __deepcopy__(self, memo):
        """
        Return a new duplicate Shape
        """

        return Square(copy.deepcopy(self.side))

    def __str__(self):
        """
        Print the Square
        """

        formatStr = (
            "{:<" +
            str(Shape.WIDTH_LABEL) +
            "}:{:>" +
            str(Shape.WIDTH_VALUE) +
            ".4f}\n"
        )

        return (
            super(Square, self).__str__() +
            formatStr.format("Side", self.side) +
            formatStr.format("Perimeter", self.perimeter()) +
            formatStr.format("Area", self.area())
        )


if __name__ == "__main__":
    s = Square(7)

    print(s)

File: 06-Java-Shapes/Example-1/src/main/java/shapes/Square.java
// Thomas Kennedy
// CS 330 Fall 2014

package shapes;

/**
 * A Rectangle with 4 Equal Sides
 */
public class Square extends Shape
{
    private double _side; ///< Length of One Side

    /**
     * Construct a Square with side set to 1
     */
    public Square()
    {
        super("Square");
        _side = 1;
    }

    /**
     * Construct a Square
     *
     * @param s the desired side length
     */
    public Square(double s)
    {
        super("Square");
        _side = s;
    }

    /**
     * Construct a Square
     *
     * @param src the Square to copy
     */
    public Square(Square src)
    {
        super("Square");
        this._side = src._side;
    }

    /**
     * Return the side length
     */
    public double side()
    {
        return _side;
    }

    /**
     * Modify the side length
     *
     * @param s the replacement length
     */
    public void side(double s)
    {
        _side = s;
    }

    @Override
    public double area()
    {
        return Math.pow(_side, 2.0);
    }

    @Override
    public double perimeter()
    {
        return 4 * _side;
    }

    @Override
    public Shape clone()
    {
        return new Square(this);
    }

    @Override
    public String toString()
    {
        return super.toString()
             + String.format(FMT_DBL, "Side", this._side)
             + String.format(FMT_DBL, "Perimeter", this.perimeter())
             + String.format(FMT_DBL, "Area", this.area());
    }
}





